src/atlas/projection/detail/ProjectionUtilities.h

```diff
@@ -0,0 +1,150 @@
+    // Left or right hand system
+    double zz_fac = 1.0;
+    if (right_hand) {zz_fac = 0.0;}
+
+    double r = sqrt(xyz[XX]*xyz[XX] + xyz[YY]*xyz[YY] + xyz[ZZ]*xyz[ZZ]);
+    lonlat[LON] = atan2(xyz[YY],xyz[XX]);
+    if ( (std::abs(xyz[XX]) + std::abs(xyz[YY])) < util::Constants::esl() ) {
+      lonlat[LON] = 0.0;  // Poles:
+    }
+    lonlat[LAT] = acos(xyz[ZZ]/r) - zz_fac * M_PI/2.0;
+  }
+
+  //------------------------------------------------------------------------------------------------
+
+  static void sphericalToCartesian(double lonlat[], double xyz[], bool right_hand = false,
+                                  bool unit = false) {
+    // Radius
+    double r = util::Earth::radius();
+    if (unit) {r = 1.0;}
+
+    // Left or right hand system
+    double zz_fac = 1.0;
+    if (right_hand) {zz_fac = -1.0;}
+
+    xyz[XX] = r * cos(lonlat[LON]) * cos(lonlat[LAT]);
+    xyz[YY] = r * sin(lonlat[LON]) * cos(lonlat[LAT]);
+    xyz[ZZ] = zz_fac * -r * sin(lonlat[LAT]);
+  }
+
+  //------------------------------------------------------------------------------------------------
+
+  static void rotate3dX(double angle, double xyz[]) {
+    const double c = cos(angle);
+    const double s = sin(angle);
+    double xyz_in[3];
+    std::copy(xyz, xyz+3, xyz_in);
+    xyz[YY] =  c*xyz_in[YY] + s*xyz_in[ZZ];
+    xyz[ZZ] = -s*xyz_in[YY] + c*xyz_in[ZZ];
+  };
+
+  //------------------------------------------------------------------------------------------------
+
+  static void rotate3dY(double angle, double xyz[]) {
+    const double c = cos(angle);
+    const double s = sin(angle);
+    double xyz_in[3];
+    std::copy(xyz, xyz+3, xyz_in);
+    xyz[XX] = c*xyz_in[XX] - s*xyz_in[ZZ];
+    xyz[ZZ] = s*xyz_in[XX] + c*xyz_in[ZZ];
+  };
+
+  //------------------------------------------------------------------------------------------------
+
+  static void rotate3dZ(double angle, double xyz[]) {
+    const double c = cos(angle);
+    const double s = sin(angle);
+    double xyz_in[3];
+    std::copy(xyz, xyz+3, xyz_in);
+    xyz[XX] =  c*xyz_in[XX] + s*xyz_in[YY];
+    xyz[YY] = -s*xyz_in[XX] + c*xyz_in[YY];
+  };
+
+  //------------------------------------------------------------------------------------------------
+
+  static void vect_cross(double xyz1[], double xyz2[], double xyzp[]) {
+    // Perform cross products of 3D vectors: xyzp = xyz1 X xyz2
+    xyzp[XX] = xyz1[YY] * xyz2[ZZ] - xyz1[ZZ] * xyz2[YY];
+    xyzp[YY] = xyz1[ZZ] * xyz2[XX] - xyz1[XX] * xyz2[ZZ];
+    xyzp[ZZ] = xyz1[XX] * xyz2[YY] - xyz1[YY] * xyz2[XX];
+  }
+
+  //------------------------------------------------------------------------------------------------
+
+  static void mirror_latlon(double lonlat2[], double lonlat3[], double lonlat1[],
+                            double lonlat4[]) {
+    // Given the "mirror" as defined by (lon1, lat1), (lon2, lat2), and center
+    // of the sphere, compute the mirror image of (lon0, lat0) as  (lon3, lat3)
+
```

Comment:
// Given the "mirror" as defined by (lon2, lat2), (lon3, lat3), and center
// of the sphere, compute the mirror image of (lon1, lat1) as  (lon4, lat4) ?

Reply:
Comment deleted